Update documentation for 2.0.0 release

fabiobrz opened the pull request:
SSIA

#### Short description of what this resolves:

Documentation update

#### Changes proposed in this pull request:

- Mainly removing the Boot2Docker and Docker-machine mentions
- Minor fixes here and there as they were spotted


Fixes #1344

gaol reviewed (commented):
Sorry  for late, several comments :)

fabiobrz commented on docs/configuration.adoc:
```diff
@@ -50,43 +50,39 @@ Let's see valid attributes:
 
 |machineDriver
-|Sets the machine driver in case you are using _docker-machine_, _Cube_ will create a machine using this driver. This parameter is mandatory when docker-machine is not installed.
+|Sets the machine driver in case you are using _docker-machine_. This parameter is mandatory when docker-machine is not installed.
```
Done, and submitted an issue for removing the remaining references in the functional and test code: https://github.com/arquillian/arquillian-cube/issues/1346

fabiobrz commented on docs/compose.adoc:
```diff
@@ -7,11 +7,11 @@ It is important to note that this is not a docker-compose implementation but onl
+Last thing, in case you define a command that is not implemented in Arquillian Cube, this command will be ignored (no exception will be thrown), but a log line will be printed notifying this situation. Please it is really important that if this happens you open a bug so we can add support for them. Although this warning we will try to maintain aligned with the latest docker-compose format.
 
-Let's see how you can rewrite previous HelloWorld example with Tomcat to be used using docker-compose format.
+Let's see how you can rewrite previous HelloWorld example with Tomcat, using docker-compose format.
```
Thanks @gaol, should be fixed by #1347 now. Would you mind to file an issue for supporting the compose-spec as suggested, so yu can deal with details you might have on your mind?

fabiobrz commented:
> Sorry for late, several comments :)

Thanks indeed for your review @gaol! I provided some feedback here, and addressed your comments via #1347, would you mind having a look?
